### DISCUSSION TITLE
Source code for Labs from "Building a Batch Application with Spring Batch"

### DISCUSSION THREAD

**Discussion by Sax388** [spring-projects/spring-batch#4572] [Q&A] [answered]
I really liked the course: https://spring.academy/courses/building-a-batch-application-with-spring-batch. It would be helpful to get access to the repository where the code (for the final project) is being hosted.

I looked through some samples in https://github.com/spring-projects/spring-batch/tree/main/spring-batch-samples but they still use discouraged annotations like `@EnableBatchProcessing` (https://github.com/search?q=repo%3Aspring-projects%2Fspring-batch%20%40EnableBatchProcessing&type=code).

**Comment by fmbenhassine** [comment, marked as answer]
Hi @Sax388 , great to hear you liked the course! The source code of the Labs is hosted on the Tanzu learning repository here: https://github.com/vmware-tanzu-learning/course-spring-batch-essentials-code. The last commit is the one that corresponds to the final project.

Please note that using `@EnableBatchProcessing` is only discouraged in the context of a Spring Boot application (like the one developed in the course) as it disables the auto-configuration of the batch infrastructure. However, the samples in the Spring Batch repository are not based on Spring Boot (in fact, not all Spring Batch users are Spring Boot users), that's why they still use that annotation. Hopefully this clarifies the difference.

**Reply by Sax388** [reply]
Thank you @fmbenhassine ! Althought I suppose there's a large intersect between Spring Boot and Batch users I really had the misconception that it was literally all of them 😄 .
